core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```diff
@@ -292,6 +292,7 @@ public void clearTemporalResponses() {
      */
     @Override
     public void ping(final PingListener listener, final TimeValue duration) {
+        final List<DiscoveryNode> sortedNodesToPing = buildNodesToPing();
```

Comment:
can we unify the resolvedDiscoveryNodes logic with the buildNodesToPing ? They are similar in the sense that they don't change during a ping cycle.